src/containers/MainPage/SignOnModal/index.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { setUsername } from 'redux/login/actions';
+
+import { Modal, Input, message, Icon } from 'antd';
+
+class SignOnModal extends React.Component {
+  state = {
+    visible: false,
+    username: '',
+  };
+
+  componentDidMount() {
+    if (!this.signedOn()) this.showModal();
+  }
+
+  signedOn = () => {
+    if (localStorage.getItem('username')) return true;
```

Comment:
Here we can use `username` from reducer `login -> username` to check if we already signed on.